nav2_gps_waypoint_follower_demo/nav2_gps_waypoint_follower_demo/interactive_waypoint_follower.py

```diff
@@ -0,0 +1,43 @@
+import rclpy
+from rclpy.node import Node
+from nav2_simple_commander.robot_navigator import BasicNavigator
+from geometry_msgs.msg import PointStamped
+from nav2_gps_waypoint_follower_demo.utils.gps_utils import latLonYaw2Geopose
+
+
+class InteractiveGpsWpCommander(Node):
+    """
+    ROS2 node to send gps waypoints to nav2 received from mapviz's point click publisher
+    """
+
+    def __init__(self):
+        super().__init__(node_name="gps_wp_commander")
+        self.navigator = BasicNavigator("basic_navigator")
+
+        self.mapviz_wp_sub = self.create_subscription(
+            PointStamped, "/clicked_point", self.mapviz_wp_cb, 1)
+
+    def mapviz_wp_cb(self, msg: PointStamped):
+        """
+        clicked point callback, sends received point to nav2 gps waypoint follower if its a geographic point
+        """
+        if msg.header.frame_id != "wgs84":
+            self.get_logger().warning(
+                "Received point from mapviz that ist not in wgs84 frame. This is not a gps point and wont be followed")
+            return
+
+        self.navigator.waitUntilNav2Active(localizer='robot_localization')
+        wp = [latLonYaw2Geopose(msg.point.y, msg.point.x)]
+        self.navigator.followGpsWaypoints(wp)
```

Comment:
Does this preempt cleanly if the user sends multiple goals without waiting between them for completion?

Reply:
It does
https://github.com/ros-planning/navigation2_tutorials/assets/71234974/06808937-8de3-41da-9621-245b45ec69cd

Reply:
OK!